src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -185,7 +185,8 @@ public ResponseEntity<Participation> startParticipation(@PathVariable Long exerc
         if (exercise instanceof ProgrammingExercise) {
             // fetch additional objects needed for the startExercise method below
             var programmingExercise = programmingExerciseRepository.findByIdWithTemplateAndSolutionParticipationElseThrow(exercise.getId());
-            if (!featureToggleService.isFeatureEnabled(Feature.ProgrammingExercises) || !isAllowedToParticipateInProgrammingExercise(programmingExercise, null)) {
+            if (!featureToggleService.isFeatureEnabled(Feature.ProgrammingExercises)
+                    || (!authCheckService.isAtLeastInstructorForExercise(exercise, user) && !isAllowedToParticipateInProgrammingExercise(programmingExercise, null))) {
```

Comment:
according to the documentation, editors can use this features as well.
https://docs.artemis.cit.tum.de/admin/accessRights/#programming-exercises

Could you please double check this? If this is correct, then we should allow this for editors as well.
Please also adda comment above that shortly explains this in natural language

Reply:
After reviewing the documentation, it's confirmed that editors can use these features too. I will also add a short explanation comment.